### PR TITLE
LPS-164770 Poshi Automation | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-164770

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/FormContainerWithPermissions.testcase [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+				var objectId = JSONObject.getObjectId(objectName = "PurchaseOrder");
+
+				Permissions.definePermissionViaJSONAPI(
+					resourceAction = "VIEW",
+					resourceName = "com_liferay_object_web_internal_object_definitions_portlet_ObjectDefinitionsPortlet_${objectId}",
+					roleTitle = "Roles Regrole Name",
+					roleType = "regular");
+			}
+
+			task ("Add a martes page template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/FormContainerWithPermissions.testcase:122]: typo

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/FormContainerWithPermissions.testcase [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+			JSONGroup.deleteGroupByName(groupName = "Test Site Name");
+
+			ObjectAdmin.deleteAllCustomObjectsViaAPI();
+		}
+	}
+
+	@description = "This is a test for LPS-164125. Form Components should available on Masters Page Template for user with custom permissions."
+	@priority = "4"
+	test MapEntityToFormContainerInMastersPageTemplateWithPermission {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/FormContainerWithPermissions.testcase:74]: MapEntityToFormContainerInMastersPageTemplateAsUserWithViewPermissionForCustomObject

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/FormContainerWithPermissions.testcase [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+			AssertSelectedLabel(
+				key_fieldLabel = "Content Type",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Purchase Order");
+		}
+	}
+
+	@description = "This is a test for LPS-164125. Form Components should not be available on Display Page Template for user with custom permissions."
+	@priority = "4"
+	test UserWithPermissionShouldNotSeeFormComponentsOnDisplayPageTemplate {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/FormContainerWithPermissions.testcase:175]: This case is already covered by CannotViewFormComponentsOnDisplayPageTemplate. The owner has view permission for custom object by default.

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind updating according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/FormContainerWithPermissions.testcase (hunk 3 above)
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/outoftheboxfragments/formcomponents/FormContainerWithPermissions.testcase:175]: All right. I'll remove it. Thank!

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind updating according to comments? Thanks

Hi @Tim-Cao, PR updated! Could you please have a look at? Thanks!

**Comment by Tim-Cao** [comment]
LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**